doc/modules/model_evaluation.rst

```diff
@@ -2254,87 +2245,102 @@ function::
-bias in sample variance of y.
+  >>> from sklearn.metrics import max_error
+  >>> y_true = [3, 2, 7, 1]
+  >>> y_pred = [9, 2, 7, 1]
+  >>> max_error(y_true, y_pred)
+  6
 
-In the particular case where the true target is constant, the :math:`R^2` score is
-not finite: it is either ``NaN`` (perfect predictions) or ``-Inf`` (imperfect
-predictions). Such non-finite scores may prevent correct model optimization
-such as grid-search cross-validation to be performed correctly. For this reason
-the default behaviour of :func:`r2_score` is to replace them with 1.0 (perfect
-predictions) or 0.0 (imperfect predictions). If ``force_finite``
-is set to ``False``, this score falls back on the original :math:`R^2` definition.
+The :func:`max_error` does not support multioutput.
 
-Here is a small example of usage of the :func:`r2_score` function::
+.. _explained_variance_score:
 
-  >>> from sklearn.metrics import r2_score
-  >>> y_true = [3, -0.5, 2, 7]
-  >>> y_pred = [2.5, 0.0, 2, 8]
-  >>> r2_score(y_true, y_pred)
-  0.948...
-  >>> y_true = [[0.5, 1], [-1, 1], [7, -6]]
-  >>> y_pred = [[0, 2], [-1, 2], [8, -5]]
-  >>> r2_score(y_true, y_pred, multioutput='variance_weighted')
-  0.938...
-  >>> y_true = [[0.5, 1], [-1, 1], [7, -6]]
-  >>> y_pred = [[0, 2], [-1, 2], [8, -5]]
-  >>> r2_score(y_true, y_pred, multioutput='uniform_average')
-  0.936...
-  >>> r2_score(y_true, y_pred, multioutput='raw_values')
-  array([0.965..., 0.908...])
-  >>> r2_score(y_true, y_pred, multioutput=[0.3, 0.7])
-  0.925...
-  >>> y_true = [-2, -2, -2]
-  >>> y_pred = [-2, -2, -2]
-  >>> r2_score(y_true, y_pred)
-  1.0
-  >>> r2_score(y_true, y_pred, force_finite=False)
-  nan
-  >>> y_true = [-2, -2, -2]
-  >>> y_pred = [-2, -2, -2 + 1e-8]
-  >>> r2_score(y_true, y_pred)
-  0.0
-  >>> r2_score(y_true, y_pred, force_finite=False)
-  -inf
+Explained variance score
+-------------------------
 
-.. topic:: Example:
+The :func:`explained_variance_score` computes the `explained variance
+regression score <https://en.wikipedia.org/wiki/Explained_variation>`_.
 
-  * See :ref:`sphx_glr_auto_examples_linear_model_plot_lasso_and_elasticnet.py`
-    for an example of R² score usage to
-    evaluate Lasso and Elastic Net on sparse signals.
+If :math:`\hat{y}` is the estimated target output, :math:`y` the corresponding
+(correct) target output, and :math:`Var` is `Variance
+<https://en.wikipedia.org/wiki/Variance>`_, the square of the standard deviation,
+then the explained variance is estimated as follow:
+
+.. math::
+
+  explained\_{}variance(y, \hat{y}) = 1 - \frac{Var\{ y - \hat{y}\}}{Var\{y\}}
+
+The best possible score is 1.0, lower values are worse.
+
+.. topic:: Link to :ref:`r2_score`
+
+    The difference between the explained variance score and the :ref:`r2_score`
+    is that when the explained variance score does not account for
+    systematic offset in the prediction. For this reason, the
+    :ref:`r2_score` should be prefered in general.
```

Comment:
This is an important note / warning. Should it be made even more concrete with an example?

```python
>>> from sklearn.datasets import load_diabetes
>>> from sklearn.linear_model import Ridge
>>> from sklearn.metrics import r2_score, explained_variance_score
>>> X, y = load_diabetes(return_X_y=True)
>>> m = Ridge().fit(X, y)
>>> r2_score(y, m.predict(X))
0.4512...
>>> r2_score(y, m.predict(X) - 42.)
0.1537...
>>> explained_variance_score(y, m.predict(X))
0.4512...
>>> explained_variance_score(y, m.predict(X) - 42.)
0.4512...
```

I am not sure that all readers will really understand what "does not account for systematic offset in the prediction" mean in practice.